packages/codegen-ui-react/lib/__tests__/__snapshots__/studio-ui-codegen-react-forms.test.ts.snap

```diff
@@ -727,7 +728,7 @@ function ArrayField({
     setSelectedBadgeIndex(undefined);
   };
   const addItem = async () => {
-    if (!!currentFieldValue && !hasError) {
+    if ((!!currentFieldValue || currentFieldValue === false) && !hasError) {
```

Comment:
I could be totally wrong here but isn't `!!currentFieldValue` already going to cover `currentFieldValue === false`?
Did you mean to accomplish something else, like `!!currentFieldValue && currentFieldValue !== false` or `currentFieldValue === undefined`?